adapters/AppLovin/AppLovinAdapter/GADMAdapterAppLovinRewardBasedVideoAd.m

```diff
@@ -0,0 +1,263 @@
+
++ (NSString *)adapterVersion
+{
+    return kGADMAdapterAppLovinVersion;
+}
+
++ (Class<GADAdNetworkExtras>)networkExtrasClass
+{
+    return [GADMAdapterAppLovinExtras class];
+}
+
+- (instancetype)initWithRewardBasedVideoAdNetworkConnector:(id<GADMRewardBasedVideoAdNetworkConnector>)connector
+{
+    if ( !connector )
+    {
+        return nil;
+    }
+    
+    self = [super init];
+    if ( self )
+    {
+        self.connector = connector;
+    }
+    return self;
+}
+
+- (void)setUp
+{
+    [self log: @"Attempting to initialize SDK"];
+
+    self.sdk = [GADMAdapterAppLovinUtils sdkForCredentials: _connector.credentials];
+    
+    if ( self.sdk )
+    {
+        [self log: @"Successfully initialized SDK"];
+        [self.connector adapterDidSetUpRewardBasedVideoAd: self];
+    }
+    else
+    {
+        [self log: @"Failed to initialize SDK"];
+        NSError *error = [NSError errorWithDomain: kGADMAdapterAppLovinErrorDomain
+                                             code: kGADErrorMediationAdapterError
+                                         userInfo: @{NSLocalizedFailureReasonErrorKey : @"AppLovin rewarded adapter unable to initialize."}];
+        [self.connector adapter: self didFailToSetUpRewardBasedVideoAdWithError: error];
+    }
+}
+
+- (void)requestRewardBasedVideoAd
+{
+    [self log: @"Requesting AppLovin rewarded video"];
+    
+    self.placement = [GADMAdapterAppLovinUtils placementFromCredentials: _connector.credentials];
+    
+    if ( self.incent.readyForDisplay )
+    {
+        [_connector adapterDidReceiveRewardBasedVideoAd: self];
+    }
+    else
+    {
+        [self.incent preloadAndNotify: self];
+    }
+}
+
+- (void)presentRewardBasedVideoAdWithRootViewController:(UIViewController *)viewController
+{
+    if ( self.incent.readyForDisplay )
+    {
+        self.reward = nil;
+        self.fullyWatched = NO;
+        
+        GADMAdapterAppLovinExtras *extras = _connector.networkExtras;
+        
+        if ( extras && extras.muteAudio )
+        {
+            self.sdk.settings.muted = YES;
+        }
+        
+        if ( self.placement )
+        {
+            [self.incent showOver: [UIApplication sharedApplication].keyWindow placement: self.placement andNotify: self];
+        }
+        else
+        {
+            [self.incent showAndNotify: self];
+        }
+    }
+    else
+    {
+        [self log: @"No ad available or attempted to show rewarded video before one was loaded"];
+        
+        NSError *error = [NSError errorWithDomain: kGADMAdapterAppLovinErrorDomain
+                                             code: [GADMAdapterAppLovinUtils toAdMobErrorCode: kALErrorCodeUnableToRenderAd]
+                                         userInfo: @{NSLocalizedFailureReasonErrorKey : @"Adaptor requested to display a rewarded video before one was loaded"}];
+        
+        [_connector adapter: self didFailToSetUpRewardBasedVideoAdWithError: error];
```

Comment:
Similar to interstitial, please send ad opened and closed callbacks here as well.